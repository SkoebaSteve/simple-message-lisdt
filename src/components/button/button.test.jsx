/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import toJson from 'enzyme-to-json'
import Button from './'

const button = {
  label: 'important',
  className: 'special',
}

it('Has the default values.', () => {
  const wrapper = shallow(<Button {...button} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('Fires a click event', () => {
  const onButtonClick = sinon.spy()
  const wrapper = shallow(<Button {...button} onClick={onButtonClick} />)
  wrapper.find('button').simulate('click')
  expect(onButtonClick).toHaveProperty('callCount', 1)
})
