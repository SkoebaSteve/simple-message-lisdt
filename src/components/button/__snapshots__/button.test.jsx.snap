// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has the default values. 1`] = `
<button
  className="button special"
>
  important
</button>
`;
